src/Cpp/spatialstructures/src/graph.cpp

```diff
@@ -1268,45 +1300,70 @@ namespace HF::SpatialStructures {
 			// If the node id provided does not exist in the value map...add it.
 			node_attr_value_map[id] = score;
 
-			// Update this iterator so it can be used in the next code block
-			node_attr_value_map_it = node_attr_value_map.find(id);
 		}
+		else
+		{
+			// Otherwise, the key-value pair exists, and we can set the value to score
+			node_attr_value_map_it->second = score;
+		}
+	}
 
-		// Will be used to assess whether it is floating point, or not
-		std::string found_attr_value = node_attr_value_map_it->second;
-
-		// Let's determine the data type of score:
-		bool score_is_floating_pt = is_floating_type(score);
+	void Graph::AddNodeAttributeFloat(int id, const std::string& attribute, const float score)
```

Comment:
the main caller names the second parameter "name" instead of attribute.  I think this should be consistent throughout.